haigha/transports/socket_transport.py

```diff
@@ -29,26 +34,33 @@ def connect(self, (host, port), klass=socket.socket):
         Connect assuming a host and port tuple.
         '''
         self._host = "%s:%s" % (host, port)
-        self._sock = klass()
-        self._sock.setblocking(True)
-        self._sock.settimeout(self.connection._connect_timeout)
-        if self.connection._sock_opts:
-            for k, v in self.connection._sock_opts.iteritems():
-                family, type = k
-                self._sock.setsockopt(family, type, v)
-        infos = socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP)
-        exc = socket.error("getaddrinfo returns an empty list")
-        for _family, _socktype, _proto, _canonname, sockaddr in infos:
+
+        for info in socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP):
```

Comment:
`getaddrinfo` can raise `socket.gaierror`. Since it appears that the new code wishes to raise `ConnectionError` when the `connect` method fails, for consistency, it should raise `ConnectionError` in this case, too. For example:

```
try:
    socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP)
except Exception as ex:
    self.connection.logger.exception('getaddrinfo failed')
    raise ConnectionError('getaddrinfo failed: %r' % (ex,))


for info in infos:
    family, socktype, proto, _, sockaddr = info
    . . . 
```